include/sys/param.h

```diff
@@ -43,6 +43,12 @@
 #  define MAX(a,b)      (((a) > (b)) ? (a) : (b))
 #endif  /* MAX */
 
+/* Macros for number of items. */
+
+#ifndef nitems
+#  define nitems(_a)    (sizeof(_a) / sizeof(0[(_a)]))
+#endif /* nitems */
```

Comment:
`ARRAY_SIZE` is also used by Linux, so we could simply define both:
https://elixir.bootlin.com/linux/latest/source/include/linux/kernel.h#L55

```suggestion
/* Macros for number of items. */

#ifndef ARRAY_SIZE
#  define ARRAY_SIZE    (sizeof(_a) / sizeof(0[(_a)]))
#endif
#ifndef nitems
#  define nitems(_a)    (sizeof(_a) / sizeof(0[(_a)]))
#endif /* nitems */
```

Reply:
@gustavonihei It would be great but see the objection voiced by @xiaoxiang781216 inhttps://github.com/apache/nuttx/pull/8414#discussion_r1096233335

Reply:
Some third-party libraries are directly using ARRAY_SIZE(). I made similar changes before this PR, if enable libuv, ci will report redefined warning:

```bash
libuv/test/task.h:77: warning: "ARRAY_SIZE" redefined
   77 | #define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
      | 
In file included from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv/unix.h:41,
                 from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv.h:66,
                 from libuv/test/test-uname.c:22:
/home/archer/code/nuttx/n8/incubator-nuttx/include/sys/param.h:47: note: this is the location of the previous definition
   47 | #  define ARRAY_SIZE(_a) (sizeof(_a) / sizeof(0[(_a)]))
      | 
In file included from libuv/test/test-walk-handles.c:23:
libuv/test/task.h:77: warning: "ARRAY_SIZE" redefined
   77 | #define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
      | 
In file included from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv/unix.h:41,
                 from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv.h:66,
                 from libuv/test/test-walk-handles.c:22:
/home/archer/code/nuttx/n8/incubator-nuttx/include/sys/param.h:47: note: this is the location of the previous definition
   47 | #  define ARRAY_SIZE(_a) (sizeof(_a) / sizeof(0[(_a)]))
      | 
In file included from libuv/test/test-watcher-cross-stop.c:23:
libuv/test/task.h:77: warning: "ARRAY_SIZE" redefined
   77 | #define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
      | 
In file included from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv/unix.h:41,
                 from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv.h:66,
                 from libuv/test/test-watcher-cross-stop.c:22:
/home/archer/code/nuttx/n8/incubator-nuttx/include/sys/param.h:47: note: this is the location of the previous definition
   47 | #  define ARRAY_SIZE(_a) (sizeof(_a) / sizeof(0[(_a)]))
      | 
In file included from libuv/test/test-udp-try-send.c:23:
libuv/test/task.h:77: warning: "ARRAY_SIZE" redefined
   77 | #define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
      | 
In file included from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv/unix.h:41,
                 from /home/archer/code/nuttx/n8/apps/system/libuv/libuv/include/uv.h:66,
                 from libuv/test/test-udp-try-send.c:22:
/home/archer/code/nuttx/n8/incubator-nuttx/include/sys/param.h:47: note: this is the location of the previous definition
   47 | #  define ARRAY_SIZE(_a) (sizeof(_a) / sizeof(0[(_a)]))
      | 
CC:  stdio/lib_tmpnam.c In file included from libuv/test/run-tests.c:34:
libuv/test/task.h:77: warning: "ARRAY_SIZE" redefined
   77 | #define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
      | 

```

https://github.com/libuv/libuv/blob/55b5d88b01ef9e1101df62ee49b3f34e58877faa/src/uv-common.h#L50